test/e2e/run_env_test.go

```diff
@@ -76,6 +76,11 @@ ENV hello=world
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 		Expect(session.OutputToString()).To(ContainSubstring("world-earth"))
+
+		session = podmanTest.Podman([]string{"run", "--rm", "--env-merge", "foo=${bar}-earth", "test", "printenv", "foo"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		Expect(session.OutputToString()).To(ContainSubstring("-earth"))
```

Comment:
You should really use `Equal()` here. It is important to match exactly for such tests.